src/Core/Services/MetadataProviders/Converters/TypeConverter.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.DataApiBuilder.Core.Services.MetadataProviders.Converters
+{
+    /// <summary>
+    /// This is a converter to serialize and deserialize an object of Type : System.Type
+    /// For example, the ColumnDefiniton object's default value property of type System.Type.
+    /// </summary>
+    public class TypeConverter : JsonConverter<Type>
+    {
+        public override Type Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.String)
+            {
+                throw new JsonException();
+            }
+
+            string typeName = reader.GetString()!;
+            return Type.GetType(typeName)!;
+        }
+
+        public override void Write(Utf8JsonWriter writer, Type value, JsonSerializerOptions options)
+        {
+            writer.WriteStringValue(value.FullName);
```

Comment:
Earlier, I observed you used AssemblyQualifiedName but here you are using fullName, could you elaborate in comment here why fullname is chosen?

Reply:
AssemblyQualified Name brings in version name, whole path and project path. In one of the comments from Sean he suggested if we can use the FullName instead which is shorter. I tested with TypeConverter and feels like for .Net types we should be good to use Full Name instead.

the same doesnot apply to DatabaseObject, it failed and gives error on "Unable to find DatabaseTable".

Reply:
added documentation like discussed offline